core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -503,16 +514,18 @@ class WorkspaceService(protected val userInfo: UserInfo,
 
       // Delete Google Project
       _ <- traceWithParent("maybeDeleteGoogleProject", parentSpan)(_ =>
-        maybeDeleteGoogleProject(workspaceContext.googleProjectId, workspaceContext.workspaceVersion, userInfo) recoverWith {
-          case t:Throwable => {
-            logger.error(s"Unexpected failure deleting workspace (while deleting google project) for workspace `${workspaceName}`", t)
-            Future.failed(t)
+        if (!isAzureMcWorkspace(maybeMcWorkspace)) {
+          maybeDeleteGoogleProject(workspaceContext.googleProjectId, workspaceContext.workspaceVersion, maybeMcWorkspace, userInfo) recoverWith {
+            case t: Throwable => {
```

Comment:
We could, but I think in this case we want as broad a type as we can get since this `recoverWith` just exists to log an error before re-throwing the error and we don't want to miss any errors